File: app/src/main/java/me/imzack/app/ender/view/adapter/TypeMarkColorGridAdapter.java
package me.imzack.app.ender.view.adapter;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import me.imzack.app.ender.R;
import me.imzack.app.ender.model.bean.TypeMarkColor;
import me.imzack.app.ender.view.widget.CircleColorView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class TypeMarkColorGridAdapter extends BaseAdapter {

    private List<TypeMarkColor> mTypeMarkColorList;

    public TypeMarkColorGridAdapter(List<TypeMarkColor> typeMarkColorList) {
        this.mTypeMarkColorList = typeMarkColorList;
    }

    @Override
    public int getCount() {
        return mTypeMarkColorList.size();
    }

    @Override
    public TypeMarkColor getItem(int position) {
        return mTypeMarkColorList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        TypeMarkColor typeMarkColor = getItem(position);
        View view;
        ViewHolder viewHolder;
        if (convertView == null) {
            view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_grid_type_mark_color, parent, false);
            viewHolder = new ViewHolder(view);
            view.setTag(viewHolder);
        } else {
            view = convertView;
            viewHolder = (ViewHolder) view.getTag();
        }

        viewHolder.mColorIcon.setFillColor(Color.parseColor(typeMarkColor.getColorHex()));

        return view;
    }

    class ViewHolder {
        @BindView(R.id.ic_color)
        CircleColorView mColorIcon;

        public ViewHolder(View view) {
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/kotlin/me/imzack/app/end/view/adapter/TypeMarkColorGridAdapter.kt
package me.imzack.app.end.view.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_grid_type_mark_color.*
import me.imzack.app.end.R
import me.imzack.app.end.model.bean.TypeMarkColor

class TypeMarkColorGridAdapter(private val mTypeMarkColorList: List<TypeMarkColor>) : BaseAdapter() {

    override fun getCount() = mTypeMarkColorList.size

    override fun getItem(position: Int) = mTypeMarkColorList[position]

    override fun getItemId(position: Int) = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val (colorHex) = getItem(position)
        val view: View
        val viewHolder: ViewHolder
        if (convertView == null) {
            view = LayoutInflater.from(parent.context).inflate(R.layout.item_grid_type_mark_color, parent, false)
            viewHolder = ViewHolder(view)
            view.tag = viewHolder
        } else {
            view = convertView
            viewHolder = view.tag as ViewHolder
        }

        viewHolder.ic_color.setFillColor(Color.parseColor(colorHex))

        return view
    }

    class ViewHolder(override val containerView: View) : LayoutContainer
}
